examples_x035/usbpd_sink/usbpd.h

```diff
@@ -0,0 +1,921 @@
+	eUSBPD_DATA_MSG_BATTERY_STATUS = 0x05u,
+	eUSBPD_DATA_MSG_ALERT = 0x06u,
+	eUSBPD_DATA_MSG_GET_COUNTRY_INFO = 0x07u,
+	eUSBPD_DATA_MSG_ENTER_USB = 0x08u,
+	eUSBPD_DATA_MSG_EPR_REUEST = 0x09u,
+	eUSBPD_DATA_MSG_EPR_MODE = 0x0Au,
+	eUSBPD_DATA_MSG_SOURCE_INFO = 0x0Bu,
+	eUSBPD_DATA_MSG_REVISION = 0x0Cu,
+	eUSBPD_DATA_MSG_VENDOR_DEFINED = 0x0Fu
+} USBPD_DataMessage_e;
+
+typedef union
+{
+	uint16_t data;
+	struct
+	{
+		uint16_t MessageType : 5u; // USBPD_ControlMessage_t | USBPD_DataMessage_t
+		USBPD_PortDataRole_e PortDataRole : 1u;
+		USBPD_SpecificationRevision_e SpecificationRevision : 2u;
+		USBPD_PortPowerRole_e PortPowerRole : 1u;
+		uint16_t MessageID : 3u;
+		uint16_t NumberOfDataObjects : 3u; // 0: Control Message, >0: Data Message
+		uint16_t Extended : 1u;
+	};
+} USBPD_MessageHeader_t;
+
+typedef USBPD_MessageHeader_t USBPD_ControlMessage_t;
+static_assert( sizeof( USBPD_MessageHeader_t ) == sizeof( uint16_t ), "USBPD_MessageHeader_t size mismatch" );
+
+typedef enum PACKED
+{
+	eUSBPD_PDO_FIXED = 0,
+	eUSBPD_PDO_BATTERY = 1,
+	eUSBPD_PDO_VARIABLE = 2,
+	eUSBPD_PDO_AUGMENTED = 3,
+} USBPD_PowerDataObject_e;
+
+typedef enum PACKED
+{
+	eUSBPD_APDO_SPR_PPS = 0, // Standard Power Range Programmable Power Supply
+	eUSBPD_APDO_EPR_AVS = 1, // Extended Power Range Adjustable Voltage Supply
+	eUSBPD_APDO_SPR_AVS = 2, // Standard Power Range Adjustable Voltage Supply
+	eUSBPD_APDO_RESERVED = 3,
+} USBPD_AugmentedPDO_e;
+
+typedef enum PACKED
+{
+	eUSBPD_PEAK_CURRENT_0 = 0, /* Peak Current equals IoC */
+
+	eUSBPD_PEAK_CURRENT_1 = 1, /* 150% IoC for 1ms @ 5% duty cycle
+	                              125% IoC for 2ms @ 10% duty cycle
+	                              110% IoC for 10ms @ 50% duty cycle */
+
+	eUSBPD_PEAK_CURRENT_2 = 2, /* 200% IoC for 1ms @ 5% duty cycle
+	                              150% IoC for 2ms @ 10% duty cycle
+	                              125% IoC for 10ms @ 50% duty cycle */
+
+	eUSBPD_PEAK_CURRENT_3 = 3, /* 200% IoC for 1ms @ 5% duty cycle
+	                              175% IoC for 2ms @ 10% duty cycle
+	                              150% IoC for 10ms @ 50% duty cycle */
+} USBPD_PeakCurrent_e;
+
+typedef struct
+{
+	uint32_t data : 28u; // PDO specific data based on PDO type
+	USBPD_AugmentedPDO_e AugmentedType : 2u; // shall be eUSBPD_APDO_SPR_PPS
+	USBPD_PowerDataObject_e PDOType : 2u; // shall be eUSBPD_PDO_AUGMENTED
+} USBPD_PDOHeader_t;
+
+typedef struct
+{
+	uint32_t MaxCurrentIn10mA : 10u;
+	uint32_t VoltageIn50mV : 10u;
+	USBPD_PeakCurrent_e PeakCurrent : 2u;
+	uint32_t Reserved_22bit : 1u;
+	uint32_t EPRModeCapable : 1u;
+	uint32_t UnchunkedExtendedMessage : 1u;
+	uint32_t DualRoleData : 1u;
+	uint32_t USBCommunicationsCapable : 1u;
+	uint32_t UnconstrainedPower : 1u;
+	uint32_t USBSuspendSupported : 1u;
+	uint32_t DualRolePower : 1u;
+	USBPD_PowerDataObject_e PDOType : 2u; // shall be eUSBPD_PDO_FIXED
+} USBPD_SourceFixedSupplyPDO_t;
+
+#define FIXED_SUPPLY_FMT                 \
```

Comment:
Please just put these in-line with where they are used.  It makes it more difficult to see the forest through the trees, since these are not core to using the USBPD.